services/frontend/src/store/user/index.js

```diff
@@ -0,0 +1,109 @@
+import { api } from '@/api';
+
+const defaultState = {
+    isLoggedIn: null,
+    token: '',
+    userID: null,
+    username: null,
+    logInError: false,
+    dashboardMiniDrawer: false,
+  };
+  
+  export const userModule = {
+    state: defaultState,
+    mutations: {
+        setToken(state, payload) {
+            state.token = payload;
+        },
+        setUsername(state, payload) {
+            state.username = payload
+        },
+        setUserID(state, payload) {
+            state.userID = payload
+        },
+        setLoggedIn(state, payload) {
+            state.isLoggedIn = payload;
+        },
+        setLogInError(state, payload) {
+            state.logInError = payload;
+        },
+        setDashboardMiniDrawer(state, payload) {
+            state.dashboardMiniDrawer = payload;
+        },
+    },
+    actions: {
+        async actionLogIn(state, payload) {
+            try {
+                const response = await api.LogIn(payload)
+                const token = response.data.access_token;
+                if (token) {
+                    saveLocalToken(token);
+                    state.commit("setToken", token);
+                    state.commit("setLoggedIn", true)
+                    state.commit("setLoggedIn", false)
+                    await state.dispatch("actionGetUser"); //! this can not work, got to check this
+                } else {
+                    await state.dispatch("actionLogOut");
+                }
+            } catch (err) {
+                state.commit("setLoggedIn", true)
+                await state.dispatch("actionLogOut");
+            }
+        },
+        async actionGetUser(state) {
+            try {
+                const response = await api.getUser(state.state.token, state.state.userID)
```

Comment:
"state.state" this is a bad naming scheme

Reply:
done, changed to contex.state